tbd/template/template.go

```diff
@@ -0,0 +1,64 @@
+package template
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"path/filepath"
+)
+
+type MissingError string
+
+func (e MissingError) Error() string {
+	return fmt.Sprintf("no template exists with name '%s'", string(e))
+}
+
+type Templates map[string]*template.Template
+
+type Template func(io.Writer, interface{}) error
+
+// Get name template from the Templates collection.
+func (t Templates) Get(name string) Template {
+	tmpl := t[name]
+
+	if tmpl == nil {
+		return func(wr io.Writer, data interface{}) error {
+			return MissingError(name)
+		}
+	}
+
+	return func(wr io.Writer, data interface{}) error {
+		return tmpl.ExecuteTemplate(wr, "page", data)
+	}
+}
+
+// Parse each html/template in templateGlob with the templates in layoutGlob,
+// providing the custom funcs.
+func Parse(layoutGlob, templateGlob string, funcs template.FuncMap) (Templates, error) {
```

Comment:
This could be changed to take a single argument of `webDir` given we use the same directory structure everywhere?